File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
  public int longestConsecutive(int[] nums) {
    if (nums == null || nums.length == 0) return 0;
        HashSet<Integer> set = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            set.add(nums[i]);
        }

        int maxLength = 1, length, curr;
        for (int i = 0; i < nums.length; i++) {
            if (set.contains(nums[i] + 1) && !set.contains(nums[i] - 1)) {
                length = 1;
                curr = nums[i];
                while (set.contains(curr + 1)) {
                    length++;
                    curr++;
                }
                maxLength = Math.max(maxLength, length);
            }
        }

        return maxLength;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums)==1:
            return nums[0]
        n = len(nums)
        dp1 = [0]*(n+1)
        dp2 = [0]*(n+1)

        dp1[n-1]=nums[n-1]

        for i in range(n-2,0,-1):
            dp1[i] = max(dp1[i+1],nums[i]+dp1[i+2])
        

        dp2[n-2]=nums[n-2]
        for i in range(n-3,-1,-1):
            dp2[i] = max(dp2[i+1],nums[i]+dp2[i+2])
        

        return max(dp1[1],dp2[0])

        
